packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts

```diff
@@ -191,8 +204,23 @@ export abstract class QueueProcessingServiceBase extends Construct {
     }
     this.cluster = props.cluster || this.getDefaultCluster(this, props.vpc);
 
-    // Create the SQS queue if one is not provided
-    this.sqsQueue = props.queue !== undefined ? props.queue : new Queue(this, 'EcsProcessingQueue', {});
+    // Create the SQS queue if one is not provided, when creating SQS queue setup DLQ
+    if (props.queue) {
+      this.sqsQueue = props.queue;
+    } else {
+      this.deadLetterQueue = new Queue(this, "EcsProcessingDeadLetterQueue", {
+        retentionPeriod: Duration.days(14)
```

Comment:
I would allow the retentionPeriod to be configured and 14 seems like a magic number to me. I think it would make more sense to allow sqs to default this to 4 days (do not specify a default, allow it to pass undefined), if the user does not specify it.

Reply:
it's not a magic number, it's in fact the max retention period for a DLQ. and i think this is a fair assumption to make for a DLQ, we want to keep it for the longest possible period.  

reference to CDKs affinity to this thought process https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda/lib/function.ts#L687-L689

Reply:
Can we make this configurable?

Reply:
Can you make this configurable (as an input property)?

Reply:
i believe the lesser the configs an user has to learn and understand the better; i don't see the downsides of setting it to max value. my reasons for believing so:

* user doesn't have to pay more
* user has message for max possible time in order to deal with it
* if user is really proactive about dealing with dlq messages they'll consume and delete sooner than 14 days

Reply:
As a user I should have the freedom to change any property within the construct. Otherwise, a user would need to re-write the entire construct in order to define a different value. 

A use case would be if an organization had an internal policy for how long specific data can be retained (7 days for instance), this 14 day default would prevent users from abiding by that policy, but by making it configurable, a user can override this default.

Reply:
make sense.